File: iot-rust/src/core/runtime/state.rs
#[derive(PartialEq)]
pub enum StatusValue {
    Initial,
    Ready,
}

pub struct State {
    status: StatusValue,
}


impl State {
    pub fn new() -> Self {
        Self { status: StatusValue::Initial }
    }
    pub fn is_ready(&self) -> bool {
        self.status == StatusValue::Ready
    }

}

File: packages/iot/core/state.go
package core

// DeviceStatus représente le statut du device
type DeviceStatus string

const (
	DeviceStatusInitializing DeviceStatus = "initializing"
	DeviceStatusReady        DeviceStatus = "ready"
	DeviceStatusError        DeviceStatus = "error"
)

// State représente l'état global de l'application
type State struct {
	Status DeviceStatus
}

// NewState crée un nouvel état initial
func NewState() *State {
	return &State{
		Status: DeviceStatusInitializing,
	}
}

